zerver/lib/actions.py

```diff
@@ -2989,3 +2997,21 @@ def make_dict(row):
     streams.sort(key=lambda elt: elt["name"])
 
     return streams
+
+def process_attachments(message):
+
+    def extract_path_id(upload_string):
+        cleaned_path = upload_string.replace('/user_uploads/', '')
+        cleaned_path = re.sub('[\(\)]', '', cleaned_path)
+        return cleaned_path
+
+    p = re.compile('\(.*?\)')
+    attachment_list = p.findall(message['message'].content)
```

Comment:
The regex stored in p is used to get every pair of '(' and ')' in a non greedy manner. There should be a check in extract_path_id to remove all the useless pairs of parenthesis. Ill add that